services/QuillLMS/app/models/student_response.rb

```diff
@@ -0,0 +1,255 @@
+#  question_id                                :bigint           not null
+#  student_response_answer_text_id            :bigint
+#  student_response_directions_text_id        :bigint
+#  student_response_instructions_text_id      :bigint
+#  student_response_previous_feedback_text_id :bigint
+#  student_response_prompt_text_id            :bigint
+#  student_response_question_type_id          :bigint
+#
+# Indexes
+#
+#  idx_student_responses_on_student_response_instructions_text_id  (student_response_instructions_text_id)
+#  idx_student_responses_on_student_response_previous_feedback_id  (student_response_previous_feedback_text_id)
+#  index_student_responses_on_activity_session_id                  (activity_session_id)
+#  index_student_responses_on_question_id                          (question_id)
+#  index_student_responses_on_student_response_answer_text_id      (student_response_answer_text_id)
+#  index_student_responses_on_student_response_directions_text_id  (student_response_directions_text_id)
+#  index_student_responses_on_student_response_prompt_text_id      (student_response_prompt_text_id)
+#  index_student_responses_on_student_response_question_type_id    (student_response_question_type_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (activity_session_id => activity_sessions.id)
+#  fk_rails_...  (student_response_answer_text_id => student_response_answer_texts.id)
+#  fk_rails_...  (student_response_directions_text_id => student_response_directions_texts.id)
+#  fk_rails_...  (student_response_instructions_text_id => student_response_instructions_texts.id)
+#  fk_rails_...  (student_response_previous_feedback_text_id => student_response_previous_feedback_texts.id)
+#  fk_rails_...  (student_response_prompt_text_id => student_response_prompt_texts.id)
+#  fk_rails_...  (student_response_question_type_id => student_response_question_types.id)
+#
+class StudentResponse < ApplicationRecord
+  belongs_to :activity_session
+  belongs_to :question
+  belongs_to :student_response_answer_text
+  belongs_to :student_response_directions_text
+  belongs_to :student_response_instructions_text
+  belongs_to :student_response_previous_feedback_text
+  belongs_to :student_response_prompt_text
+  belongs_to :student_response_question_type
+
+  has_one :student_response_extra_metadata, dependent: :destroy
+
+  has_one :student_response_concept_result, dependent: :destroy
+  has_one :concept_result, through: :student_response_concept_result
+
+  has_many :student_responses_concepts, dependent: :destroy
+  has_many :concepts, through: :student_responses_concepts
+
+
+  validates_exclusion_of :correct, in: [nil]
+  validates_presence_of :attempt_number, :question_number
+
+  # This is a list of keys from the old ConceptResults records
+  # that this model knows how to normalize.  Any keys that are
+  # fed into create_from_json that aren't from this list,
+  # will be stashed in a related StudentResponseExtraMetadata
+  # record
```

Comment:
If we're relegating the extra metadata to a separate table, why not throw it away entirely?

Reply:
This is a fair question.  I think a good chunk of this is a sense of paranoia, but on the more practical side of things, one of the things we're planning to do is test this extensively in staging.  Doing so should allow us to use the `extra_metadadata` table to ACTUALLY inventory the keys that we don't currently support.  The hope is that there will be a lot fewer rows in that table than in the full `concept_results` table because most of the `ConceptResults` records should transfer into `StudentResponse` records that have no `extra_metadata`.  We may, once we take a look at what sorts of data ends up in `extra_metadata`, drop the table.  We just want to make sure we look at it first.